cmd/wallet-web/src/pages/chapi/Signup.vue

```diff
@@ -0,0 +1,207 @@
+    }
+
+    await this.loadOIDCUser()
+
+    try {
+      await this.refreshUserMetadata()
+    } catch (e) {
+      console.warn("first time login: ignore", e)
+    }
+
+    if (this.getCurrentUser()) {
+      await this.finishOIDCLogin()
+      await this.getAgentInstance().store.flush()
+      this.handleSuccess()
+      return
+    }
+    if (this.$cookies.isKey('registerSuccess')) {
+      this.registered = true;
+    }
+
+    this.stopLoading()
+    this.loading = false
+  },
+  components: {
+    BeatLoader,
+  },
+  data() {
+    return {
+      providers:[],
+      hubOauthProvider: this.hubURL(),
+      statusMsg: '',
+      loading: true,
+      registered: false,
+      messageNum: 0,
+      messages: [
+        "Getting Started..",
+        "This could take a minute.",
+        "Please do not refresh the page.",
+        "Please wait...",
+      ]
+    };
+  },
+  methods: {
+    ...mapActions({
+      loadUser: 'loadUser',
+      loadOIDCUser: 'loadOIDCUser',
+      refreshUserMetadata: 'refreshUserMetadata',
+      startUserSetup: 'startUserSetup',
+      completeUserSetup: 'completeUserSetup'
+    }),
+    ...mapGetters(['getCurrentUser', 'getAgentOpts', 'serverURL', 'hubURL']),
+    ...mapGetters('agent', {getAgentInstance: 'getInstance'}),
+    beginOIDCLogin: async function () {
+      window.location.href = this.serverURL() + "/oidc/login"
+    },
+
+    // Fetching the providers from hub-auth
+    fetchProviders: async function() {
+      await axios.get(this.hubURL() +'/oauth2/providers')
+          .then(response => {
+            this.providers = response.data.authProviders
+          })
+    },
+
+    sleep(ms) {
+      return new Promise(resolve => setTimeout(resolve, ms));
+    },
+    finishOIDCLogin: async function () {
+      let user = this.getCurrentUser()
+
+      let registrar = new RegisterWallet(this.$polyfill, this.$webCredentialHandler, this.getAgentInstance(),
+          this.getAgentOpts())
+
+      try {
+        if (!user.metadata) {
+          this.startUserSetup()
+
+          // first time login, register this user
+          registrar.register(user.username, this.completeUserSetup)
+        }
+
+        await this.getAgentInstance().store.flush()
+
+        await registrar.installHandlers(user.username)
+      } catch (e) {
+        this.handleFailure(e)
+      }
+    },
+    handleSuccess() {
+      this.$router.push(this.redirect);
+    },
+    handleFailure(e) {
+      console.error("login failure: ", e)
+      this.statusMsg = e.toString()
+    },
+    loginDevice: async function () {
+      await this.deviceLogin.login();
+    },
+    startLoading() {
+      this.intervalID = setInterval(() => {
+        this.messageNum++;
+        this.messageNum %= this.messages.length;
+      }, 3000);
+    },
+    stopLoading() {
+      clearInterval(this.intervalID);
+    },
+
+  }
+}
+
+</script>
+<style scoped>
+.login-button {
+  text-transform: none !important; /*For Lower case use lowercase*/
```

Comment:
Do we need `important` to override some Vue-Material styles?

Reply:
This will go any way once we have our themes defined. Yes for now this was the easiest way to do